pandas/core/arrays/base.py

```diff
@@ -400,6 +400,36 @@ def dropna(self):
 
         return self[~self.isna()]
 
+    def shift(self, periods=1):
+        # type: (int) -> ExtensionArray
+        """
+        Shift values by desired number.
+
+        Newly introduced missing values are filled with
+        ``self.dtype.na_value``.
+
+        Parameters
+        ----------
+        periods : int, default 1
+            The number of periods to shift. Negative values are allowed
+            for shifting backwards.
+
+        Returns
+        -------
+        shifted : ExtensionArray
+        """
+        if periods == 0:
+            return self.copy()
```

Comment:
Thoughts on this? non-zero periods will necessarily be a copy.

Reply:
Although `Series.shift(0)` doesn't make a copy, so let's follow that?

Reply:
I personally find that strange behaviour to not copy then (it would be more logical to know that the result is always a copy, regardless the value of periods IMO)

Reply:
Opened https://github.com/pandas-dev/pandas/issues/22397 for that. Will leave this as is then.